contributing-device-support.md

```diff
@@ -1,120 +1,207 @@
-# Contributing new board support to resin.io
+# Contributing New Board Support to Resin.io
 
-Pre-requisites: yocto BSP layer for your particular board, supplying the BSP (board support package). Should be compatible to the yocto releases resin currently supports.
+Pre-requisites: a [Yocto](https://www.yoctoproject.org) Board Support Package (BSP) layer for your particular board. It should be compatible to the Yocto releases resinOS supports.
 
-Every repository used to build a resin.io host Operating System is typically named resin-`<board-family>`. For example, consider the [resin-intel][resin-intel repo] repository which can be used to build a resin.io image for
-the Intel NUC boards.
+Repositories used to build resinOS host Operating System (OS) are typically named resin-`<board-family>`. For example, consider [resin-raspberrypi](https://github.com/resin-os/resin-raspberrypi) which is used for building the OS for [Raspberryi Pi](https://raspberrypi.org), or [resin-intel][resin-intel repo] repository which can be used to build a resin.io image for the Intel NUC boards.
 
-### Repository breakout
------------------------
+Contributing support for a new board consist of creating a a Yocto package that includes:
 
-The resin-`<board-family>` repository uses git submodules for including required subprojects.
+* general hardware support for the specific board,
+* the resinOS-specific software features,
+* deployment-specific features (i.e. settings to create SD card images or self-flashing images)
 
-The root directory shall contain 2 directory entries: a "layers" directory and the [resin-yocto-scripts][resin-yocto-scripts] git submodule.
-This submodule is typically using latest master branch.
-The root directory also includes the following files: CHANGELOG.md, LICENSE, README.md, VERSION and one or more files (depending on the number of supported boards)
-named `<yocto-machine-name>`.coffee (e.g. [intel-corei7-64.coffee][intel-corei7-64 coffee]) which correspond to each of the boards this repository will add support for.
+The following documentations walks you through the steps of creating such a Yocto package. Because of the substantial difference between the hardware of many boards, this document provides general directions, and often it might be helpful to see the examples of already supported boards. The list of the relevant repositories is found the end of this document.
```

Comment:
Should be:

"The list of the relevant repositories is found at the end of this document."